streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1108,7 +1090,20 @@ private void subscribeConsumer() {
         if (topologyMetadata.usesPatternSubscription()) {
             mainConsumer.subscribe(topologyMetadata.sourceTopicPattern(), rebalanceListener);
```

Comment:
Should we throw an error here that pattern subscriptions are not supported?

Reply:
That is a good idea!